unicodetools/src/main/java/org/unicode/text/UCD/VersionedSymbolTable.java

```diff
@@ -0,0 +1,426 @@
+                            new String[] {"Ll", "Lm", "Lo", "Lt", "Lu"},
+                            UcdPropertyValues.General_Category_Values.Cased_Letter,
+                            new String[] {"Ll", "Lt", "Lu"},
+                            UcdPropertyValues.General_Category_Values.Mark,
+                            new String[] {"Mc", "Me", "Mn"},
+                            UcdPropertyValues.General_Category_Values.Number,
+                            new String[] {"Nd", "Nl", "No"},
+                            UcdPropertyValues.General_Category_Values.Punctuation,
+                            new String[] {"Pc", "Pd", "Pe", "Pf", "Pi", "Po", "Ps"},
+                            UcdPropertyValues.General_Category_Values.Symbol,
+                            new String[] {"Sc", "Sk", "Sm", "So"},
+                            UcdPropertyValues.General_Category_Values.Separator,
+                            new String[] {"Zl", "Zp", "Zs"});
+
+    /**
+     * Similar to iup.getProperty(UcdProperty.General_Category).getSet(propertyValue), but takes the
+     * groupings into account. Implements both unary-query-expression for a General_Category alias and
+     * binary-query-expression with a property-value where the queried property is General_Category.
+     */
+    private UnicodeSet getGeneralCategorySet(IndexUnicodeProperties iup, String propertyValue) {
+        var gc = iup.getProperty(UcdProperty.General_Category);
+        for (var entry : COARSE_GENERAL_CATEGORIES.entrySet()) {
+            final var aliases = entry.getKey().getNames().getAllNames();
+            if (aliases.stream().anyMatch(a -> UnicodeProperty.equalNames(propertyValue, a))) {
+                UnicodeSet result = new UnicodeSet();
+                for (var value : entry.getValue()) {
+                    gc.getSet(value, result);
+                }
+                return result;
+            }
+        }
+        return gc.getSet(propertyValue);
+    }
+
+    @Override
+    public boolean applyPropertyAlias(String beforeEquals, String afterEquals, UnicodeSet result) {
+        result.clear();
+        String leftHandSide = beforeEquals;
+        String propertyPredicate = afterEquals;
+        boolean interiorlyNegated = false;
+        int posNotEqual = beforeEquals.indexOf('≠');
+        // TODO(egg): We cannot distinguish \p{X=} from \p{X} in this API, both give us an empty
+        // string as afterEquals.  This is an @internal API, so we could change it to pass null in
+        // the unary case.
+        if (posNotEqual >= 0) {
+            propertyPredicate =
+                    afterEquals.length() == 0
+                            ? beforeEquals.substring(posNotEqual + 1)
+                            : beforeEquals.substring(posNotEqual + 1) + "=" + afterEquals;
+            leftHandSide = beforeEquals.substring(0, posNotEqual);
+            interiorlyNegated = true;
+        }
+        if (interiorlyNegated) {
```

Comment:
nit: you don't need this variable if you do this processing inside the `if (posNotEqual >= 0)` block

Reply:
Yes, but the name is also acting as documentation (it matches a defined term in the spec).